### DISCUSSION TITLE
PR commit shows as Unverified

### DISCUSSION THREAD

**Discussion by fmunteanu** [community/community#147710] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

I enabled the [Vigilant](https://docs.github.com/github/authenticating-to-github/displaying-verification-statuses-for-all-of-your-commits) mode on my profile, but for some reason, all Git commits show a unverified (PR [example](https://github.com/axivo/k3s-cluster/pull/341)):

![image](https://github.com/user-attachments/assets/743149b8-6ee6-4352-ba9b-7fc082044455)

The SSH key is uploaded and functional, since I'm able to push commits from my local repository:

![Screenshot 2024-12-21 at 9 01 33 PM](https://github.com/user-attachments/assets/e6c98e59-6ff5-4b11-921e-743f07b7ee80)

I'm not sure what I miss. Steps I did on my Mac:

```
brew install gpg2
git config --global gpg.format ssh
git config --global user.signingkey $HOME/.ssh/id_ed25519.pub
git config --global commit.gpgsign true
git config --global tag.gpgSign true
cat > $HOME/.ssh/allowed_signers <<EOF
fmunteanu@users.noreply.github.com ssh-ed25519 AAAAC3N[redacted] fmunteanu@users
EOF
git config --global gpg.ssh.allowedSignersFile $HOME/.ssh/allowed_signers
git config --global log.showSignature true
```

In VSCode I also enabled `"git.enableCommitSigning": true`, to sign the commits. I don't understand why GitHub is asking me to upload the SSH public key, since is already there. Thank you for letting me know what I miss.

**Comment by davevad93** [comment]
Hi @fmunteanu , 

Try adding the `-S` flag to git commit, for example:

 ```bash
git commit -S -m "commit message"
```

Check if by signing your commit the problem persists.

**Reply by fmunteanu** [reply]
The problem is not the switch, `-S` is equivalent to:

```
git config --global commit.gpgsign true
```

The issue is GitHub telling me the public key is not added to my account, when in fact it is. The commit is signed properly.

For GitHub action, is a different story. I have no idea what steps to take.

**Comment by fmunteanu** [comment, marked as answer]
The problem was, which is not well documented, you need to add an additional SSH key for signing. You can use the same public key but set it as signing key:

![image](https://github.com/user-attachments/assets/c0915b3b-5499-470d-b5c6-38637eee09ed)

In other words, you end-up with two identical keys, one for authentication and one for signing commits.

**Reply by davevad93** [reply]
Ah that's right, I totally forgot that, yes you do need a double `SSH Key` one for authentication and another one for signing. Glad you figured it out @fmunteanu 👍
